src/wng_building/station_river_creator.py

```diff
@@ -1,3 +1,124 @@
+import copy
+
+from src.data_handling.data_handler import DataHandler
+
+
 class StationRiverCreator:
-    def __init__(self):
-        pass
+    """
+    Class for creating the following data structures: stations, rivers, completed rivers
+    """
+    def __init__(self, data_handler: DataHandler):
+        """
+        Constructor.
+        :param DataHandler data_handler: a DataHandler instance
+        """
+        self.data_handler = data_handler
+
+        self.stations = None
+        self.rivers = None
+        self.completed_rivers = None
+
+    def run(self) -> None:
+        """
+        Run function. Gets stations, rivers and completed rivers.
+        """
+        self.data_handler.run()
+
+        self.stations = self.create_stations()
+
+        self.rivers = self.create_rivers()
+
+        self.completed_rivers = self.create_completed_rivers()
+
+    def create_stations(self) -> dict:
```

Comment:
Ez menthet közvetlenül a tagváltozóba, ha később nem akarjuk kívül is meghívni. Így csak a hívás lesz a `run` metódusban.